support-workers/src/main/scala/com/gu/support/workers/lambdas/UpdateSupporterProductData.scala

```diff
@@ -0,0 +1,125 @@
+
+class UpdateSupporterProductData(serviceProvider: ServiceProvider)
+  extends SubsetServicesHandler[SendAcquisitionEventState, Unit, SendThankYouEmailState](
+    serviceProvider,
+    _.sendThankYouEmailState
+  ) {
+
+  def this() = this(ServiceProvider)
+
+  override protected def subsetHandler(
+    state: SendThankYouEmailState,
+    requestInfo: RequestInfo,
+    context: Context,
+    services: Services
+  ): FutureHandlerResult =
+    updateSupporterProductData(state, services.catalogService, services.supporterDataDynamoService)
+      .map(_ => HandlerResult((), requestInfo))
+
+  def updateSupporterProductData(
+    state: SendThankYouEmailState,
+    catalogService: CatalogService,
+    supporterDataDynamoService: SupporterDataDynamoService
+  ) = {
+    getSupporterRatePlanItemFromState(state, catalogService) match {
+      case Some(supporterRatePlanItem) => supporterDataDynamoService.writeItem(supporterRatePlanItem)
+      case None => Future.failed(
+          new RuntimeException(s"Unable to update the SupporterProductData dynamo table because we couldn't create a SupporterRatePlanItem from state: $state")
+        )
+    }
+  }
+
+  def getSupporterRatePlanItemFromState(state: SendThankYouEmailState, catalogService: CatalogService): Option[SupporterRatePlanItem] =
+    state match {
+      case SendThankYouEmailContributionState(user, product, _, _, subscriptionNumber) =>
+        catalogService
+          .getProductRatePlan(Contribution, product.billingPeriod, NoFulfilmentOptions, NoProductOptions)
+          .map(productRatePlan =>
+            supporterRatePlanItem(
+              subscriptionName = subscriptionNumber,
+              identityId = user.id,
+              productRatePlanId = productRatePlan.id,
+              productRatePlanName = "support-workers added Contribution",
+            ))
+      case SendThankYouEmailDigitalSubscriptionDirectPurchaseState(user, product, _, _, _, _, subscriptionNumber) =>
+        catalogService
+          .getProductRatePlan(DigitalPack, product.billingPeriod, NoFulfilmentOptions, NoProductOptions)
+          .map(productRatePlan =>
+            supporterRatePlanItem(
+              subscriptionName = subscriptionNumber,
+              identityId = user.id,
+              productRatePlanId = productRatePlan.id,
+              productRatePlanName = "support-workers added Digital Subscription",
+            ))
+      case SendThankYouEmailDigitalSubscriptionCorporateRedemptionState(user, product, _, subscriptionNumber) =>
+        catalogService
+          .getProductRatePlan(DigitalPack, product.billingPeriod, NoFulfilmentOptions, NoProductOptions, Corporate)
+          .map(productRatePlan =>
+            supporterRatePlanItem(
+              subscriptionName = subscriptionNumber,
+              identityId = user.id,
+              productRatePlanId = productRatePlan.id,
+              productRatePlanName = "support-workers added Gift Digital Subscription",
```

Comment:
actualy corporate.  can we just `.describe` the product instead or toString on the State class simpleName? as the one below is also wrong

Reply:
describe is a good idea